streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -692,8 +690,7 @@ public boolean process(final long wallClockTime) {
                 processorContext.timestamp(),
                 processorContext.headers()
             );
-            final Optional<RecordMetadata> recordMetadata = Optional.ofNullable(processorContext.recordContext());
-            maybeMeasureLatency(() -> currNode.process(toProcess, recordMetadata), time, processLatencySensor);
+            maybeMeasureLatency(() -> currNode.process(toProcess), time, processLatencySensor);
```

Comment:
I think this is a proof of code simplicity that recordContext is unnecessarily passing around here :)

Reply:
Yeah, I agree. I think this is much simpler for both the implementation and the API. Sorry I didn't see it at first ;)